src/pybind/mgr/dashboard/frontend/src/app/shared/components/pwd-expiry-notification/pwd-expiry-notification.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit } from '@angular/core';
+
+import { forkJoin as observableForkJoin } from 'rxjs';
+
+import { UserFormModel } from '../../../core/auth/user-form/user-form.model';
+import { SettingsService } from '../../api/settings.service';
+import { UserService } from '../../api/user.service';
+import { AuthStorageService } from '../../services/auth-storage.service';
+
+@Component({
+  selector: 'cd-pwd-expiry-notification',
+  templateUrl: './pwd-expiry-notification.component.html',
+  styleUrls: ['./pwd-expiry-notification.component.scss']
+})
+export class PwdExpiryNotificationComponent implements OnInit {
+  alertType: string;
+  expiryDays: number;
+  levels = {};
+  private warningLevel = 'USER_PWD_EXPIRY_WARNING_1';
+  private dangerLevel = 'USER_PWD_EXPIRY_WARNING_2';
+
+  constructor(
+    private authStorageService: AuthStorageService,
+    private userService: UserService,
+    private settingService: SettingsService
+  ) {}
+
+  ngOnInit() {
+    this.userService.get(this.authStorageService.getUsername()).subscribe((user: UserFormModel) => {
```

Comment:
When login as a non-administrator user (e.g. cluster-manager), a 403 page is displayed and the banner is not shown. `/user` endpoint can be reached by administrator roles only. Maybe we need to adjust this or find some way to return the expiry info for non-administrator users.